pkg/webhook/webhookserviceadapter.go

```diff
@@ -0,0 +1,69 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"github.com/seqeralabs/staticreg/pkg/db"
+	"log/slog"
+)
+
+type ServiceAdapter struct {
+	Logger *slog.Logger
+}
+
+func NewServiceAdapter(log *slog.Logger) *ServiceAdapter {
+	return &ServiceAdapter{
+		Logger: log,
+	}
+}
+
+func (a *ServiceAdapter) SavePullEvent(ctx context.Context, event *DistributionEvent) error {
+	if db.Pool == nil {
+		a.Logger.Debug("Skipping pull event save: Database pool is not active.")
+		return nil
+	}
+
+	repoName := event.Target.Repository
+	tag := event.Target.Tag
+	digest := event.Target.Digest
+	architecture := event.GetArchitecture()
+	pullDate := event.Timestamp.UTC().Format("2006-01-02") // Format as DATE (YYYY-MM-DD) in UTC
+
+	a.Logger.Debug("Processing pull event",
+		"repository", repoName,
+		"tag", tag,
+		"digest", digest,
+		"architecture", architecture,
+		"date", pullDate)
+
+	// Use UPSERT (INSERT ... ON CONFLICT) to increment pull count
+	// If the combination of (pull_date, repo_name, tag, digest, architecture) exists, increment pull_count
+	// Otherwise, insert a new record with pull_count = 1
+	query := `
+        INSERT INTO container_pull_metrics (pull_date, repo_name, tag, digest, architecture, pull_count, updated_at)
+        VALUES ($1, $2, $3, $4, $5, 1, NOW())
+        ON CONFLICT (pull_date, repo_name, tag, digest, architecture)
+        DO UPDATE SET
+            pull_count = container_pull_metrics.pull_count + 1,
+            updated_at = NOW()`
```

Comment:
This approach works because postgres will use row level locks: if an image with the same `pull_date, repo_name, tag, digest, architecture` is pulled many times concurrently, we might have performance issues because each update has to wait for the previous one to commit. We should keep an eye on this situation just to double check this approach works with the shape of our pull traffic.
If this is a problem, we can look into storing pull events individually and roll them up either at query time or in a daily batch job